Tiltfile

```diff
@@ -388,6 +392,104 @@ def prepare_all():
+
+    template_dirs = settings.get("template_dirs", {
+        "docker": ["./test/infrastructure/docker/templates"],
+    })
+
+    for provider, provider_dirs in template_dirs.items():
+        for template_dir in provider_dirs:
+            template_list = [filename for filename in listdir(template_dir) if os.path.basename(filename).endswith("yaml")]
+            for filename in template_list:
+                deploy_templates(filename, provider, substitutions)
+
+def deploy_templates(filename, provider, substitutions):
+    # validate filename exists
+    if not os.path.exists(filename):
+        fail(filename + " not found")
+
+    os.environ["NAMESPACE"] = substitutions.get("NAMESPACE", "default")
+    os.environ["KUBERNETES_VERSION"] = substitutions.get("KUBERNETES_VERSION", "v1.24.0")
+    os.environ["CONTROL_PLANE_MACHINE_COUNT"] = substitutions.get("CONTROL_PLANE_MACHINE_COUNT", "1")
+    os.environ["WORKER_MACHINE_COUNT"] = substitutions.get("WORKER_MACHINE_COUNT", "3")
+
+    basename = os.path.basename(filename)
+    if basename.endswith(".yaml"):
+        if basename.startswith("clusterclass-"):
+            template_name = basename.replace("clusterclass-", "").replace(".yaml", "")
+            deploy_clusterclass(template_name, provider, filename)
+        elif basename.startswith("cluster-template-"):
+            template_name = basename.replace("cluster-template-", "").replace(".yaml", "")
+            deploy_cluster_template(template_name, provider, filename)
+
+def deploy_clusterclass(template_name, provider, filename):
+    apply_clusterclass_cmd = "cat " + filename + " | " + envsubst_cmd + " | " + kubectl_cmd + " apply -f - && echo \"ClusterClass created from\'" + filename + "\', don't forget to delete\n\""
+    delete_clusterclass_cmd = kubectl_cmd + " delete clusterclass " + template_name + ' --ignore-not-found=true; echo "\n"'
+
+    local_resource(
+        name = template_name,
+        cmd = apply_clusterclass_cmd,
+        auto_init = False,
+        trigger_mode = TRIGGER_MODE_MANUAL,
+        labels = [provider + "-clusterclasses"],
+    )
+
+    cmd_button(
+        template_name + ":apply",
+        argv = ["sh", "-c", apply_clusterclass_cmd],
+        resource = template_name,
+        icon_name = "note_add",
+        text = "Apply ClusterClass",
+    )
+
+    cmd_button(
+        template_name + ":delete",
+        argv = ["sh", "-c", delete_clusterclass_cmd],
+        resource = template_name,
+        icon_name = "delete_forever",
+        text = "Delete ClusterClass",
+    )
+
+def deploy_cluster_template(template_name, provider, filename):
+    apply_cluster_template_cmd = "CLUSTER_NAME=" + template_name + "-$RANDOM; " + clusterctl_cmd + " generate cluster $CLUSTER_NAME --from " + filename + " > ./.tiltbuild/" + template_name + "; cat ./.tiltbuild/" + template_name + " | " + kubectl_cmd + " apply -f - && echo \"Cluster '$CLUSTER_NAME' created, don't forget to delete\n\""
```

Comment:
Q: Should we maybe pipe it directly from clusterctl to kubectl?
I realize that it's useful for debugging, but on the other side we are just collecting more and more files in .tiltbuild and there is no garbage collection

If we want to be able to see the YAML, maybe we can just additionally pipe it to stdout so we can see it in the web UI?

Reply:
Yeah we should be able to do that!